File: domain/src/main/java/no/nav/melosys/domain/dokument/felles/Periode.java
package no.nav.melosys.domain.dokument.felles;

import java.time.LocalDate;

import no.nav.melosys.domain.ErPeriode;

public class Periode implements ErPeriode {

    private LocalDate fom;

    private LocalDate tom;

    public Periode () {
    }

    public Periode (LocalDate fom, LocalDate tom) {
        this.fom = fom;
        this.tom = tom;
    }

    @Override
    public LocalDate getFom() {
        return fom;
    }

    @Override
    public LocalDate getTom() {
        return tom;
    }

    @Override
    public String toString() {
        return new StringBuilder().append(fom).append(" → ").append(tom).toString();
    }

}

File: domain/src/main/kotlin/no/nav/melosys/domain/dokument/inntekt/Periode.kt
package no.nav.melosys.domain.dokument.inntekt

import no.nav.melosys.domain.ErPeriode
import no.nav.melosys.domain.MaybePeriode
import no.nav.melosys.domain.SimpleErPeriodeAdapter
import no.nav.melosys.domain.toErPeriode
import java.time.LocalDate

data class Periode(
    override var fom: LocalDate? = null,
    override var tom: LocalDate? = null
) : MaybePeriode {
    // Deprecated: Use toErPeriode() extension function instead
    @Deprecated("Use toErPeriode() extension function from MaybePeriode interface", ReplaceWith("this.toErPeriode()"))
    fun tilErPeriode(): ErPeriode? = toErPeriode()

    override fun toString() = "${fom ?: "null"} → ${tom ?: "null"}"
}
